src/Lib/Canvas.vala

```diff
@@ -39,8 +36,10 @@ public class Akira.Lib.Canvas : Goo.Canvas {
             return _edit_mode;
         }
         set {
-            _edit_mode = value;
-            set_cursor_by_edit_mode ();
+            if (_edit_mode != value) {
+                _edit_mode = value;
+                set_cursor_by_edit_mode ();
```

Comment:
Since you're calling the `set_cursor_by_edit_mode ()` method here, you don't need to call it anywhere else as it will be automatically triggered whenever the edit mode changes.
Right now, if you debug the method, you'll see it gets called constantly at every mouse movement.
Leave the call only here and remove it everywhere else.

Reply:
This is generally true, the main issue there is that I need to call this function when the user clicks because I need to update the cursor in PAN mode when holding. 

To overcome this situation (that I see more as a workaround or bad coding, at least) I think we can have 2 different ´edit_mode´s, one is MODE_PAN and the other MODE_PANNING. The first is only triggered by pressing space on the keyboard, the second is when you are pressing space and clicking or clicking with the MIDDLE_MOUSE_BUTTON. What do you think?

Reply:
Yes, I like it.
That removes the necessity of having a `holding` variable, right?
I think it might be cleaner.

Reply:
Negative, sir! The `holding` variable is still necessary for other tiny things (like hovering, object moving and I think something else too). I'm going for this approach then!